src/app/model/ChessEngine.ts

```diff
@@ -119,4 +136,31 @@ export class ChessEngine implements ChessModel {
   public possibleMoves(position: Square): Square[] {
     return this.pieceMovesNotCausingAllyKingCheckmate(position);
   }
+
+  private kingWasChecked(chosenPiece: Piece): KingWasChecked | undefined {
+    const kingsSide = chosenPiece.side === Side.WHITE ? Side.BLACK : Side.WHITE;
+    const kingPosition = this.kingPosition(this.board, kingsSide);
+    // poniższe brzydkie i mi się nie podoba, ale na razie działa - możecie podpowiedzieć zmiany :)
+    let king;
+    if (kingPosition) {
+      king = this.board.onPositionPiece(kingPosition);
+    }
+    const isKingChecked = this.isKingChecked(this.board, kingsSide);
+    return isKingChecked && king instanceof King && kingPosition
+      ? {
+          eventType: 'KingWasChecked',
+          king: king,
+          onSquare: kingPosition,
+        }
+      : undefined;
+  }
+
+  private kingWasUnchecked(pieceWasMoved: PieceWasMoved): KingWasUnchecked | undefined {
```

Comment:
Tutaj jest błąd logiczny. Nie musi się ruszyć król, żeby przestał być "szachowany". Może inna bierka go zasłonić :) 
Proponuje zrobic metode do `onKingWasChecked(event: KingWasChecked)` i zapisywac w stanie gry szachowanego króla. Jeśli król był szachowany wcześniej to teraz po ruchu sprawdzasz czy jest dalej szachowany i tylko jeśli był i nie jest to generujesz ten event :)

Reply:
Ewentualnie chcac jakos ladnie (bez ifów na event) zrobić te metody on, możesz sprobować podobne podejście jak tutaj opisałem dla widoku:
https://github.com/nowakprojects/CodersCamp2020.Project.TypeScript.Chess/pull/77#discussion_r575702240

Reply:
racja, nareperuję :)